chain/x/zoracle/client/rest/query.go

```diff
@@ -13,93 +13,96 @@ import (
 	"github.com/bandprotocol/d3n/chain/x/zoracle/internal/types"
 )
 
+func newRequestQueryInfo(
+	ctx context.CLIContext, id string, queryRequest types.RequestQuerierInfo,
+) (RequestQueryInfo, error) {
+	var request RequestQueryInfo
+
+	request.OracleScriptID = queryRequest.Request.OracleScriptID
+	request.Calldata = queryRequest.Request.Calldata
+	request.RequestedValidators = queryRequest.Request.RequestedValidators
+	request.SufficientValidatorCount = queryRequest.Request.SufficientValidatorCount
+	request.ExpirationHeight = queryRequest.Request.ExpirationHeight
+	request.IsResolved = queryRequest.Request.IsResolved
+	request.RawDataRequests = queryRequest.RawDataRequests
+
+	request.Result = queryRequest.Result
+
+	// Get request detail
+	searchRequest, err := utils.QueryTxsByEvents(
```

Comment:
Can we somehow abstract "finding the transaction of this" as a separate function. We will need to reuse this a lot.